### PR TITLE
rule-parser: ignore duplicated msg keyword

### PR DESCRIPTION
**Pull request by norg**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/2074) ticket:

Describe changes:

- This fixes the leak resulted by duplicated msg keyword
- This also fails with -T as requested by previous PR

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

- PR norg-pcap: https://buildbot.openinfosecfoundation.org/builders/norg-pcap/builds/49
- PR norg: https://buildbot.openinfosecfoundation.org/builders/norg/builds/51

### CODE REVIEW

File: src/detect-msg.c [hunk 1]
```diff
@@ -103,7 +103,13 @@ static int DetectMsgSetup (DetectEngineCtx *de_ctx, Signature *s, const char *ms
         }
     }
 
-    s->msg = SCStrdup(str);
+    if (s->msg != NULL) {
+        SCLogError(SC_ERR_INVALID_RULE_ARGUMENT, "Duplicated 'msg:' keyword detected!");
+        goto error;
+    }
+    else {
```
Comment (inliniac) [inline src/detect-msg.c]: unnecessary else here, the if branch exits through the jump to 'error'

Comment (norg) [inline src/detect-msg.c]: I added the goto afterwards, you're right, will fix it

**Comment by inliniac** [comment]
I did a slight update of the error message (start with no capital, no ! at the end).

Also for future PRs please don't update an existing PR.

**Comment by inliniac** [comment]
Merged in #2862

**Pull request closed without merging**